services/src/main/scala/cromwell/services/metadata/impl/archiver/ArchiveMetadataSchedulerActor.scala

```diff
@@ -94,6 +93,13 @@ class ArchiveMetadataSchedulerActor(archiveMetadataConfig: ArchiveMetadataConfig
     }
   }
 
+  private def getGcsPathForMetadata(id: WorkflowId): Future[Path] =  {
+    val bucket = archiveMetadataConfig.bucket
+    getRootWorkflowId(id.toString).map((rootWorkflowId: Option[String]) => {
```

Comment:
I have a suspicion that this separate `getRootWorkflowId` might not be necessary. The `id` here comes from `lookupNextWorkflowToArchive`, which is unpacking a `WorkflowQuerySuccess` case class. I think that case class should also have a root workflow ID which we can possibly extract at the same time as the ID and use both here without needing to go back to the DB